pkg/clone/clone.go

```diff
@@ -0,0 +1,152 @@
+package clone
+
+import (
+	"encoding/json"
+	"fmt"
+
+	//nolint:staticcheck
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"path"
+	"path/filepath"
+)
+
+type ScriptStruct struct {
+	path      string
+	arguments []string
+}
+
+type Clone struct {
+	URL string
+}
+
+type Response struct {
+	CID string
+}
+
+func NewCloneClient() (*Clone, error) {
+	return &Clone{
+		URL: "",
+	}, nil
+}
+
+func (cl *Clone) RunShellScript(scriptPath string, args []string) (string, error) {
+	scriptArgs := []string{}
+	scriptArgs = append(scriptArgs, scriptPath)
+	scriptArgs = append(scriptArgs, args...)
+
+	if len(args) > 0 {
+		last := scriptArgs[len(scriptArgs)-1]
+		path, err := ioutil.TempDir("", path.Base(last))
+		scriptArgs[len(scriptArgs)-1] = path
+		if err != nil {
+			fmt.Println(err)
+		}
+		return path, nil
+	}
+
+	x := fmt.Sprintf("/bin/bash %v", scriptArgs)
```

Comment:
We can't assume Bash will be installed on all compute nodes.

Reply:
```
package main

import (
	"fmt"
	"os/exec"
	"runtime"
	"strings"
)

func main() {
	var cmd *exec.Cmd
	if runtime.GOOS == "windows" {
		cmd = exec.Command("cmd", "/C", "echo %COMSPEC%")
	} else {
		cmd = exec.Command("sh", "-c", "echo $SHELL")
	}
	out, err := cmd.Output()
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	shell := strings.TrimSpace(string(out))
	fmt.Println("Current shell:", shell)
}

```

will this work ?